src/Mvc/Mvc.NewtonsoftJson/test/NewtonsoftJsonOutputFormatterTest.cs

```diff
@@ -46,6 +49,81 @@ public void Constructor_UsesSerializerSettings()
             Assert.Same(serializerSettings, jsonFormatter.SerializerSettings);
         }
 
+        [Fact]
+        public async Task MvcJsonOptionsAreUsedToSetBufferThresholdFromServices()
+        {
+            // Arrange
+            var person = new User() { FullName = "John", age = 35 };
+            Stream writeStream = null;
+            var outputFormatterContext = GetOutputFormatterContext(person, typeof(User), writerFactory: (stream, encoding) =>
+            {
+                writeStream = stream;
+                return StreamWriter.Null;
+            });
+
+            var services = new ServiceCollection()
+                    .AddOptions()
+                    .Configure<MvcNewtonsoftJsonOptions>(o =>
+                    {
+                        o.OutputFormatterMemoryBufferThreshold = 1;
+                    })
+                    .BuildServiceProvider();
+
+            outputFormatterContext.HttpContext.RequestServices = services;
+
+            var settings = new JsonSerializerSettings
+            {
+                ContractResolver = new CamelCasePropertyNamesContractResolver(),
+                Formatting = Formatting.Indented,
+            };
+            var expectedOutput = JsonConvert.SerializeObject(person, settings);
+            var jsonFormatter = new NewtonsoftJsonOutputFormatter(settings, ArrayPool<char>.Shared, new MvcOptions());
+
+            // Act
+            await jsonFormatter.WriteResponseBodyAsync(outputFormatterContext, Encoding.UTF8);
+
+            // Assert
+            Assert.IsType<FileBufferingWriteStream>(writeStream);
+
+            // HACK: This we're trying to verify the settings are applied
+            var field = typeof(FileBufferingWriteStream).GetField("_memoryThreshold", BindingFlags.NonPublic | BindingFlags.Instance);
```

Comment:
We can consider exposing this API as well so that we can remove this private reflection.

cc @Tratcher

Reply:
Sure, go ahead and add `public int MemoryThreshold { get; }` to avoid the reflection.

Reply:
Done